packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -639,11 +643,21 @@ export class RemotedFluidDataStoreContext extends FluidDataStoreContext {
                     throw new Error(`Invalid snapshot format version ${snapshotFormatVersion}`);
                 }
                 this.pkg = pkgFromSnapshot;
+
+                // If there is no isRootDataStore in the attributes blob, set it to true. This will ensure that
+                // data stores in older documents are not garbage collected incorrectly. This may lead to additional
+                // roots in the document but they won't break.
+                isRootStore = isRootDataStore ?? true;
             }
 
+            assert(
```

Comment:
I do not understand the purpose of IF check on line 626. But clearly this code assumes there might be no tree, and if so, this assert would fire. Could you please spend some time to learn about the intention of this code? If it's old check that is no longer valid, then I'd rather assert much higher, and assert that tree is no null. If it can happen, then we need proper default for such case (likely true, but it's worth understanding the problem before we take that path).

Interesting - does it pass UTs / manual testing with this assert? If yes, then likely using true as default is rather safe

Reply:
You are right. It will fail if there is no snapshot in the attach message. This falls into the same scenario which Arin pointed out in the other comment. Looks like we need to support backwards compat for this for now. Even if we don't, we should not do it here.
Anyway, I am using a separate blob to store the isRootDataStore but the same scenario can happen there too. So, I have removed this assert and defaulted the value to true.